### DISCUSSION TITLE
Problema al enviar el task 1

### DISCUSSION THREAD

**Discussion by cristianpenagos** [rramosp/ai4eng.v1#306] [03 - GENERACION DE MODELOS]
buenos dias, mi codigo es el siguiente:

def make_timeseries_dataset(d, n_timesteps_lookback):
    import pandas as pd

    columnasDespl = []
    colPred = pd.DataFrame()
    for i in range(n_timesteps_lookback, -1, -1): # Num desplazamiento, Parada, Salto
      columna = d.shift(-i)
      columnasDespl.append(columna)

    df = pd.concat(columnasDespl, axis=1)

    columna_nombres = ['signal-' + str(i) for i in range(n_timesteps_lookback + 1)]
    df.columns = columna_nombres
    df = df.dropna()

    colPred["signal+1"] = d.shift(- n_timesteps_lookback - 1)
    df = pd.concat([df, colPred], axis=1)
    df = df.dropna()

    df = df.rename(columns={"signal-0":"singal"})
    
    #df.info()
    
    return df
    
    el codigo cumple con la salida esperada, sin embargo presenta el siguiente error al intentar enviar al RLXMOOC:
    
    error: ERROR EXECUTING YOUR CODE
---------------------------------
Traceback (most recent call last):
  File "<string>", line 652, in run
TypeError: make_timeseries_dataset() got an unexpected keyword argument 'signal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/content/calling_script.py", line 24, in <module>
    r = eval("onfcskyx.{}(source_functions, source_variables, '[cristian.penagos@udea.edu.co](mailto:cristian.penagos@udea.edu.co)')".format(grader_function_name), namespace)
  File "<string>", line 1, in <module>
  File "/content/onfcskyx.py", line 37, in grader_01
    return grader.run()
  File "<string>", line 710, in run
local.lib.rlxmoocapi.session.RLXMOOCException: error: ERROR GRADING YOUR SOLUTION
----------------------
make_timeseries_dataset() got an unexpected keyword argument 'signal' (set session.debug=True for tracebacks)
 (set session.debug=True for tracebacks)
 
 
 
 espero instrucciones.
 Muchas gracias

**Comment by esn8der** [comment]
A mi me muestra otro error, estas usando la variable `d` acá `d.shift(-i)` y acá `d.shift(- n_timesteps_lookback - 1)`, pero eso  no está definido en ningún lado, en vez de `d` deberías usar `signal`. Prueba cambiar eso a ver que te muestra

**Comment by Felipe-RA** [comment]
Saludos @cristianpenagos 

Como comenta @esn8der , no tienes definido `d` en ningun lado. Debes probar a usar `signal`
